### ISSUE TITLE
no-fallthrough error when there is a break / return for every case

### ISSUE THREAD

**Issue by CoderMiguel** [eslint/eslint#14265]
**Tell us about your environment**

* **ESLint Version:** v7.15.0
* **Node Version:** v14.15.1
* **npm Version:** v6.14.8

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**

babel/eslint-parser

```js
  switch (firstParam) {
    case firstParamCaseOne:
      switch (secondParam) {
        case firstParamCaseOne: return // first param case one code
        case firstParamCaseTwo: return // first param case two code
        default: return // default code
      }
    case contentViewOptions.forms:
      switch (secondParam) {
        case secondParamCaseOne:
          // second param case one code
          break;
        case secondParamCaseTwo: return // second param case two code
        default: return // second param default code
      }
    // fall through  <-- this should not be needed
    default:
      return //overall default code
  }
```

**What did you expect to happen?**

I expect to not violate the 'no-fallthrough' rule by having a break / return in every case.

**What actually happened? Please include the actual, raw output from ESLint.**
I am getting an error when using a break in secondParamCaseOne instead of a return.

**Comment by CoderMiguel** [comment]
@mdjermanovic so, add a break after the first layer case statements?

  switch (firstParam) {
    case firstParamCaseOne:
      switch (secondParam) {
        case firstParamCaseOne: return // first param case one code
        case firstParamCaseTwo: return // first param case two code
        default: return // default code
      }
    case contentViewOptions.forms:
      switch (secondParam) {
        case secondParamCaseOne:
          // second param case one code
          break;
        case secondParamCaseTwo: return // second param case two code
        default: return // second param default code
      }
      break; <-- do this instead?
    default:
      return //overall default code
  }

**Comment by mdjermanovic** [comment]
Yes, that should work if you don't want fall-through from `case contentViewOptions.forms` to `default:`.

**Comment by CoderMiguel** [comment]
thanks for the quick response and feedback, sorry for the false alarm
